### DISCUSSION TITLE
How to get the username associated with a UserID with the Dart SDK? (Social Feed)

### DISCUSSION THREAD

**Discussion by djkemmet** [pocketbase/pocketbase#5740] [Q&A] [answered]
Hi there, 

I'm building a social feed in my application, I am a first time developer so please bear me with patience. 

My table looks like this: 
![image](https://github.com/user-attachments/assets/05ed191b-34d5-40fb-bfa6-dc812a51a029)

My feed looks like this:
![image](https://github.com/user-attachments/assets/34a1d9f1-9164-4a9f-b670-7e2bd3d0f809)

and the code to produce my feed is: 

`class AppHomepageState extends ConsumerState<AppHomepage> {
  TextEditingController _archiveURLController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    // Connect / Watch our PocketBase Provider so it can be consumed in this view.
    var localPocketBaseProvider = ref.watch(PocketbaseProvider);

    TextEditingController postController = TextEditingController();

    double iconSize = 40;

    // When we build this view, we want to fetch a list of the latest posts to the social feed.
    return Scaffold(
      body: Container(
        width: MediaQuery.of(context).size.width,
        color: Colors.white,
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(
              "LATEST",
              style:
                  GoogleFonts.afacad(fontSize: 32, fontWeight: FontWeight.bold),
            ),
            SizedBox(
              height: MediaQuery.of(context).size.height / 12,
            ),
            Expanded(
              child: SizedBox(
                width: MediaQuery.of(context).size.width / 2,
                child: FutureBuilder(
                  future: localPocketBaseProvider
                      .collection("feed_posts")
                      .getList(sort: '-created'),
                  builder: (BuildContext context, dataSnapshot) {
                    if (dataSnapshot.hasData) {
                      var x = 1;
                      return ListView.builder(
                          itemCount: dataSnapshot.data!.items.length,
                          itemBuilder: (context, index) {
                            var userID = dataSnapshot
                                .data!.items[index].data['posted_by'];

                            var x = 1;
                            return Padding(
                              padding: const EdgeInsets.all(8.0),
                              child: Card(
                                child: Padding(
                                  padding: const EdgeInsets.all(8.0),
                                  child: Column(
                                    children: [
                                      // USER PROFILE PHOTO AND USERNAME
                                      Row(
                                          mainAxisAlignment:
                                              MainAxisAlignment.start,
                                          children: [
                                            Padding(
                                              padding:
                                                  const EdgeInsets.all(8.0),
                                              child: CircleAvatar(
                                                radius: 30,
                                                backgroundColor: Colors.black,
                                                child: CircleAvatar(
                                                  radius: 27,
                                                  foregroundImage: NetworkImage(
                                                      "https://media.sproutsocial.com/uploads/2022/06/profile-picture.jpeg"),
                                                ),
                                              ),
                                            ),
                                            Padding(
                                              padding:
                                                  const EdgeInsets.all(8.0),
                                              child: Text(
                                                  dataSnapshot
                                                      .data!
                                                      .items[index]
                                                      .data['posted_by'],
                                                  style: GoogleFonts.afacad(
                                                      fontSize: 28,
                                                      fontWeight:
                                                          FontWeight.bold)),
                                            )
                                          ]),
                                      // CONTENT OF THE POST CARD IN THE FEED
                                      Padding(
                                        padding: const EdgeInsets.all(8.0),
                                        child: SizedBox(
                                          width:
                                              MediaQuery.of(context).size.width,
                                          child: Text(
                                              textAlign: TextAlign.start,
                                              dataSnapshot.data!.items[index]
                                                  .data['content']),
                                        ),
                                      ),
                                      // LIKE COMMENT BUTTONS
                                      Row(children: [
                                        TextButton.icon(
                                            label: Text("Like"),
                                            icon: Icon(Icons.check),
                                            onPressed: () {}),
                                        //TextButton.icon()
                                      ])
                                    ],
                                  ),
                                ),
                              ),
                            );
                          });
                    } else {
                      return CircularProgressIndicator(
                        color: Colors.black,
                      );
                    }
                  },
                ),
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
          elevation: 5.0,
          backgroundColor: Colors.black,
          child: Icon(Icons.post_add, color: Colors.white),
          onPressed: () => showDialog(
              context: context,
              builder: (context) {
                return AlertDialog(
                  content: Container(
                    height: MediaQuery.of(context).size.height / 4,
                    width: MediaQuery.of(context).size.width / 4,
                    child: Column(
                      children: [
                        Text("NEW POST",
                            style: GoogleFonts.afacad(fontSize: 32)),
                        SizedBox(
                          height: 10,
                        ),
                        Text("What have you discovered?"),
                        TextField(controller: postController),
                      ],
                    ),
                  ),
                  actions: [
                    ElevatedButton(
                        style: ButtonStyle(
                            backgroundColor:
                                WidgetStateProperty.all(Colors.black)),
                        child: Text("POST",
                            style: GoogleFonts.afacad(
                                color: Colors.white,
                                fontWeight: FontWeight.bold)),
                        onPressed: () async {
                          try {
                            await localPocketBaseProvider
                                .collection("feed_posts")
                                .create(body: {
                              'posted_by':
                                  localPocketBaseProvider.authStore.model.id,
                              'content': postController.text
                            });
                            // Refresh "parent" widget to update feed
                            setState(() {});
                            Navigator.of(context).pop();
                          } catch (err) {
                            print(err);
                            ScaffoldMessenger.of(context)
                                .showSnackBar(SnackBar(content: Text("URROR")));
                          }
                        })
                  ],
                );
              })),
    );
  }
}`

I'm using provider to provide the instance of my PocketBase connection. What I'm trying to accomplish is have the social post card show the user's username and not the user ID.  I've tried writing a function to fetch the username in my widget's state and that fives me a future error, I cannot do it inside the LIstView builder and Inspecting my vars on debug the username doesn't seem to come back with the feed posts, just the user ID. Is there a way to update my table so that the user's username is a field returned in my request for each post? I'm kinda not sure how I address this.

**Comment by Tonku** [comment]
if you are new, have a read https://suragch.medium.com/flutter-state-management-for-minimalists-4c71a2f2f0c

**Reply by djkemmet** [reply]
Hey  @Tonku , Thanks for your time with my post. Unfortunately that post is coming back 404 not found and is not in the internet archive. is there an alternate article you could recommend?

**Comment by MichaelSowah** [comment]
https://suragch.medium.com/flutter-minimalist-state-management-weather-app-708b01417b9a

**Comment by KelvinConnor** [comment, marked as answer]
I _think_ you just need to add an 'expand' your getList call. Whatever the field name that links back to your 'user' collection  (posted_by?) can be added to the method call. The additional details will be included in the 'expand' section of the response. Check out https://pocketbase.io/docs/working-with-relations/#expanding-relations for more info.
